pkg/patterns/declarative/pkg/applier/direct.go

```diff
@@ -46,8 +46,14 @@ func (d *directApplier) NewBuilder(opt ApplierOptions) *resource.Builder {
 	return resource.NewBuilder(restClientGetter)
 }
 
-func (d *directApplier) NewFactory() cmdutil.Factory {
-	return cmdutil.NewFactory(&genericclioptions.ConfigFlags{})
+func (d *directApplier) NewFactory(opt ApplierOptions) cmdutil.Factory {
+	var configFlags genericclioptions.ConfigFlags
+	if opt.RESTConfig != nil {
```

Comment:
Might be worth a comment on why this is important.
I believe this is preventing unnecessary repeated openapi calls when a restconfig exists and has already made those calls.